packages/legacy/core/App/screens/Connection.tsx

```diff
@@ -72,102 +69,98 @@ const Connection: React.FC<ConnectionProps> = ({ navigation, route }) => {
+  //     timerRef.current = null
+  //   }
+  // }
 
-  const setModalVisible = (value: boolean) => {
-    setState((prev) => ({ ...prev, isVisible: value }))
-  }
-  const setIsInitialized = (value: boolean) => {
-    setState((prev) => ({ ...prev, isInitialized: value }))
-  }
-  const setShouldShowDelayMessage = (value: boolean) => {
-    setState((prev) => ({ ...prev, shouldShowDelayMessage: value }))
-  }
-  useEffect(() => {
-    if (autoRedirectConnectionToHome && shouldShowDelayMessage && connectionIsActive && !notificationRecord) {
-      setShouldShowDelayMessage(false)
-      setModalVisible(false)
-      navigation.getParent()?.navigate(TabStacks.HomeStack, { screen: Screens.Home })
-    }
-  }, [shouldShowDelayMessage])
-  const abortTimer = () => {
-    if (timerRef.current) {
-      clearTimeout(timerRef.current)
-      timerRef.current = null
-    }
-  }
   const onDismissModalTouched = () => {
-    setShouldShowDelayMessage(false)
-    setModalVisible(false)
+    dispatch({ shouldShowDelayMessage: false, isVisible: false })
     navigation.getParent()?.navigate(TabStacks.HomeStack, { screen: Screens.Home })
   }
-  const startTimer = () => {
-    if (!isInitialized) {
-      timerRef.current = setTimeout(() => {
-        setShouldShowDelayMessage(true)
-        timerRef.current = null
-      }, connTimerDelay)
-      setIsInitialized(true)
-    }
-  }
 
   useEffect(() => {
-    if (shouldShowDelayMessage && !notificationRecord) {
-      AccessibilityInfo.announceForAccessibility(t('Connection.TakingTooLong'))
+    // Wait for OOB proof notification to arrive.
+    if (!connection) {
+      return
     }
-  }, [shouldShowDelayMessage])
 
-  useFocusEffect(
-    useCallback(() => {
-      startTimer()
-      return () => abortTimer
-    }, [])
-  )
+    // Note: V1 connections are not required to enter the `Completed` state and may
+    // remain at `RequestSent` indefinitely. This is a known issue that is addressed
+    // in higher protocol versions.
+    if (connection.state === DidExchangeState.Completed || connection.state === DidExchangeState.RequestSent) {
+      dispatch({ connectionIsActive: true })
+      agent?.mediationRecipient.initiateMessagePickup()
 
-  useEffect(() => {
-    if (notificationRecord) {
-      switch (notificationRecord.type) {
-        case 'CredentialRecord':
-          navigation.navigate(Screens.CredentialOffer, { credentialId: notificationRecord.id })
-          break
-        case 'ProofRecord':
-          navigation.navigate(Screens.ProofRequest, { proofId: notificationRecord.id })
-          break
-        default:
-          throw new Error('Unhandled notification type')
-      }
+      // No goal code, we don't know what to expect next,
+      // navigate to the contact details.
+      navigation
+        .getParent()
+        ?.navigate(TabStacks.HomeStack, { screen: Screens.ContactDetails, params: { connectionId } })
+    }
+
+    return () => {
+      connection === undefined
```

Comment:
Just curious, what is the purpose of returning a boolean from the cleanup function?

Reply:
I'm testing to see if it clears up a memory leak waring from metro. My though was the connection or notification below change after the screen is dismissed triggering a useEffect and causing the warning. Might be a long shot but I'm not seeing it ATM.

Reply:
But isn't that a equality operator and not an assignment? I think those two vars are declared with `const` too. But if it's still working somehow we can keep it, that works for me

Reply:
Oh, yes, good catch.

Reply:
Removed. Wasn't what I was after.